### PR TITLE
Fix/configurable customerio events

### PR DESCRIPTION
**Pull request by johantjellden**

### CODE REVIEW

File: src/test/java/com/hedvig/notificationService/customerio/ConfigurableNorwaySignHackTest.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        )
+
+        every { workspaceSelector.getWorkspaceForMember(any()) } returns Workspace.NORWAY
+
+        val cut =
+            CustomerioService(workspaceSelector, stateRepository, eventCreator, clients, productPricingFacade, false)
+
+        cut.updateCustomerAttributes("someId", makeSignFromUnderwriterMap())
+
+        assertThat(stateRepository.data["someId"]?.underwriterFirstSignAttributesUpdate).isNull()
```
Comment (meletios) [inline src/test/java/com/hedvig/notificationService/customerio/ConfigurableNorwaySignHackTest.kt:37]: Was it clearer if we also verify that stateRepository.save is never called?

File: src/main/java/com/hedvig/notificationService/customerio/state/InMemoryCustomerIOStateRepository.kt [hunk 2]
```diff
@@ -16,11 +16,11 @@ class InMemoryCustomerIOStateRepository(var data: Map<String, CustomerioState> =
     override fun shouldUpdate(byTime: Instant): Collection<CustomerioState> {
         return data.values.filter {
             (it.underwriterFirstSignAttributesUpdate != null &&
-                it.underwriterFirstSignAttributesUpdate <= byTime &&
+                it.underwriterFirstSignAttributesUpdate!! <= byTime &&
```
Comment (meletios) [inline src/main/java/com/hedvig/notificationService/customerio/state/InMemoryCustomerIOStateRepository.kt:19]: Do we need to throw an exception here? or just return null?

File: src/main/java/com/hedvig/notificationService/customerio/EventHandler.kt [hunk 3]
```diff
@@ -16,25 +16,27 @@ class EventHandler(
...
-        var state = repo.findByMemberId(event.owningMemberId)
+        val state = repo.findByMemberId(event.owningMemberId)
             ?: CustomerioState(event.owningMemberId)
 
-        state = state.triggerStartDateUpdated(callTime)
-
-        if (!configuration.useNorwayHack)
-            repo.save(state.updateFirstUpcomingStartDate(event.startDate))
+        if (!configuration.useNorwayHack) {
+            repo.save(state)
```
Comment (meletios) [inline src/main/java/com/hedvig/notificationService/customerio/EventHandler.kt:23]: is this should be on line 25 instead?

File: src/main/java/com/hedvig/notificationService/customerio/CustomerioService.kt [hunk 4]
```diff
@@ -115,7 +114,7 @@ open class CustomerioService(
                 customerioState.memberId,
                 event
             )
-            this.stateRepository.save(updateFunction(customerioState))
+            this.stateRepository.save(customerioState)
```
Comment (meletios) [inline src/main/java/com/hedvig/notificationService/customerio/CustomerioService.kt:117]: 👍

File: src/main/java/com/hedvig/notificationService/customerio/state/InMemoryCustomerIOStateRepository.kt (hunk 2 above)
Comment (johantjellden) [reply src/main/java/com/hedvig/notificationService/customerio/state/InMemoryCustomerIOStateRepository.kt:19]: This is fine, this is the InMemoryRepository that we use just for testing.

File: src/test/java/com/hedvig/notificationService/customerio/ConfigurableNorwaySignHackTest.kt (hunk 1 above)
Comment (johantjellden) [reply src/test/java/com/hedvig/notificationService/customerio/ConfigurableNorwaySignHackTest.kt:37]: Maybe we should, but I think that this is enough.

**Pull request merged**